drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -177,6 +186,14 @@ static int pwm_nrf5_sw_pin_set(struct device *dev, u32_t pwm,
 pin_set_pwm_off:
 	data->map[channel].pulse_cycles = 0;
 
+	/* restart timer if another channel is still active */
```

Comment:
Can we do something better to not stop the timer in first place by checking if this the last active channel being turned off?

Reply:
Current logic first disables timer before clearing up GPIOTE and PPI, so the question would be if it is safe to clear both GPIOTE and PPI while timer is still active (I would assume so, but you are more familiar with the MCU design).

If it is fine to clear both GPIOTE and PPI without stopping timer, we can just invert the logic and stop the timer in case that is the last active channel.

Reply:
logically seems safe to keep timer running while setting up the GPIOTE and PPI, as PPI channel is enabled after the configuration in line 174. So, yes, please go ahead and implemen the change to stop timer on last channel being turned off.